provider/providers.go

```diff
@@ -147,3 +148,25 @@ func NewTwitter(p Params) Oauth1Handler {
 		},
 	})
 }
+
+// NewBattlenet makes Battle.net oauth2 provider
+func NewBattlenet(p Params) Oauth2Handler {
+	return initOauth2Handler(p, Oauth2Handler{
+		name: "battlenet",
+		endpoint: oauth2.Endpoint{
+			AuthURL:   "https://eu.battle.net/oauth/authorize",
+			TokenURL:  "https://eu.battle.net/oauth/token",
+			AuthStyle: oauth2.AuthStyleInParams,
+		},
+		scopes:  []string{},
+		infoURL: "https://eu.battle.net/oauth/userinfo",
+		mapUser: func(data UserData, _ []byte) token.User {
+			userInfo := token.User{
+				ID:   "battlenet_" + token.HashID(sha1.New(), data.Value("id")),
+				Name: data.Value("battletag"),
```

Comment:
is the lack of pictrue mapping intentional? i.e. is it supported by battlenet? If it is - we should map it

Reply:
@umputun There is no user picture in Battle.net, only pictures per user per game, some of those games don't even have an official API